x-team/auto-ui
# Add autofocus true as default to selectbox component

**Release Type:** *Non-Breaking Feature* 


Fixes https://x-team-internal.atlassian.net/browse/XP-2427

## Description

> It should be the default in the dropdown component over the whole app to auto-focus search field inside those dropdowns. This way we improve the UX and don’t ask users to do an extra click. If we click on the dropdown, we should auto-focus on the search textfield inside it if it exists. We can think of doing that on Auto UI level.



## Checklist

**Before submitting a pull request,** please make sure the following is done:



## Related PRs



branch | PR
------ | ------
XP-2427-remove-unnecessary-autofocus | [Auto](https://github.com/x-team/auto/pull/1750)

## Steps to Test or Reproduce
1- Go to the admin page
2- Confirm that all the drop-down buttons that have an input text are autofocus when is opened

Examples of such dropdowns in the app:

- Select List
- Select Fields
- Add to List



## Impacted Areas in Application
- Search form
- Add to list 



## Screenshots

![selectbox_autofocus](https://user-images.githubusercontent.com/5392202/50562142-e32f0900-0ce7-11e9-83df-cef44b1a93d4.gif)
